ansible/roles/http_proxy/tasks/main.yml

```diff
@@ -2,49 +2,80 @@
   lineinfile:
     dest: "/etc/environment"
     line: "{{item.var}}={{item.protocol}}://{{http_proxy_address}}/"
+    regexp: "{{item.var}}="
     create: yes
     state: present
   sudo: yes
   with_items:
     - {'var': 'http_proxy', 'protocol': 'http'}
     - {'var': 'https_proxy', 'protocol': 'https'}
+    - {'var': 'ftp_proxy', 'protocol': 'ftp'}
     - {'var': 'HTTP_PROXY', 'protocol': 'http'}
     - {'var': 'HTTPS_PROXY', 'protocol': 'https'}
+    - {'var': 'FTP_PROXY', 'protocol': 'ftp'}
 
 - name: No proxy for local IPs
   lineinfile:
     dest: "/etc/environment"
     line: "{{ item }}={{ groups['all']|join(',') }},relay.nic.in,smsgw.sms.gov.in,icds-cas.gov.in"
+    regexp: "{{item }}="
     create: yes
     state: present
   sudo: yes
   with_items:
     - 'no_proxy'
     - 'NO_PROXY'
 
+- name: Bower proxy
+  template:
+    src: bowerrc.j2
+    dest: "{{ item.home }}/.bowerrc"
+  become: yes
+  become_user: "{{item.user}}"
+  ignore_errors: True
```

Comment:
Sometimes user does not exist, I thinkg cchq is not created in all servers am I wrong, to be honest this also helped quite a bit while testing on my own environment.